models/account.py

```diff
@@ -103,7 +103,15 @@ def update_account(id: str, email: str, azure_ad_subject_id: str, full_name: str
             Account object or None
         """
         url = Config.ACCOUNT_STORE_API_HOST + Config.ACCOUNT_ENDPOINT.format(account_id=id)
-        cleaned_roles = [azure_ad_role_map[role] for role in roles]
+        cleaned_roles = []
+        if isinstance(roles, List):
+            cleaned_roles = [azure_ad_role_map[role] for role in roles]
+        if len(cleaned_roles) == 0:
+            current_app.logger.error(
+                f"When attempting to update account id: {id} "
+                f"with email: {email} - no roles were found"
```

Comment:
Could we make the message a bit more user friendly something like:
`account id: {id} has not been assigned any roles` and abort 403, rather than continuing?

Reply:
Alternatively we could log azure_ad_subject_id which is an Azure ref obviously?

Reply:
I didn't abort here as we actually already have an error page (with no permissions message) that users are redirected to if they don't have the right roles - so that seemed a more obvious thing to show the user maybe?

Reply:
Yeah, I think we don't follow this rule very well, but I guess better to log the account ID, so that the email address is not in the logs.

Reply:
Error route now returns 403 👍